iradix_test.go

```diff
@@ -212,6 +211,145 @@ func TestDelete(t *testing.T) {
 	}
 }
 
+func TestDeletePrefix(t *testing.T) {
+
+	type exp struct {
+		treeNodes   []string
+		prefix      string
+		expectedOut []string
+	}
+
+	//various test cases where DeletePrefix should succeed
+	cases := []exp{
+		{[]string{"", "test/test1", "test/test2", "test/test3", "R", "RA"}, "test", []string{"", "R", "RA"}},
+		{[]string{"", "test", "test/test1", "test/test2", "test/test3", "test/testAAA", "R", "RA"}, "test", []string{"", "R", "RA"}},
+		{[]string{"", "test/test1", "test/test2", "test/test3", "test/testAAA", "R", "RA"}, "test/test", []string{"", "R", "RA"}},
+		{[]string{"", "AB", "ABC", "AR", "R", "RA"}, "AR", []string{"", "AB", "ABC", "R", "RA"}},
+	}
+
+	for _, testCase := range cases {
+		verifyPrefixDelete(t, testCase.treeNodes, testCase.prefix, testCase.expectedOut)
```

Comment:
It would be good to add a short description for what each case is trying to hit, then you could do `t.Run()` to run as a sub-test and just inline the `verifyPrefixDelete` code. You could also delete an invalid prefix like "nope" as part of each case and verify it, otherwise I'd factor that out as a separate test entirely.